ph_logger.c

```diff
@@ -222,6 +229,74 @@ static char *strnchr(char *src, char ch, int len)
 	return NULL;
 }
 
+struct ph_logger_file *_search_log_file(const char *path)
+{
+	struct ph_logger_file *f, *tmp;
+	dl_list_for_each_safe(f, tmp, &ph_logger.files, struct ph_logger_file, list)
+	{
+		if (pv_str_matches(path, strlen(path), f->path, strlen(f->path)))
+			return f;
+	}
+
+	return NULL;
+}
+
+static void _save_log_file_pos(off_t pos, const char *path)
+{
+	// try to get file info from memory
+	struct ph_logger_file *f;
+	f = _search_log_file(path);
+	// if not, we create a new entry
+	if (!f) {
+		pv_log(DEBUG, "log file '%s' not yet stored in memory. Saving new file...", path);
+		f = calloc(1, sizeof(struct ph_logger_file));
+		f->path = strdup(path);
+		dl_list_add(&ph_logger.files, &f->list);
+	}
+	if (!f)
+		pv_log(ERROR, "could not initialize log file: %s", strerror(errno));
+
+	// update pos value in memory
+	f->pos = pos;
+
+	// save pos in xattr if possible by config
+	if (!pv_config_get_str(PV_STORAGE_LOGTEMPSIZE)) {
+		char value[MAX_DEC_STRING_SIZE_OF_TYPE(pos)]; // max PRI64
+		SNPRINTF_WTRUNC(value, sizeof(value), "%" PRId64, pos);
+		if (setxattr(path, PH_LOGGER_POS_XATTR, value, strlen(value), 0))
+			pv_log(WARN, "xattr could not be saved in '%s': %s", path, strerror(errno));
+	}
+}
+
+static off_t _load_log_file_pos(const char *path)
+{
+	// first, try to get pos from memory
+	struct ph_logger_file *f;
+	f = _search_log_file(path);
+	if (f)
+		return f->pos;
+
+	pv_log(DEBUG, "unknown file found in '%s'", path);
+
+	// if not in memory, try to get it from xattr if possible
+	char dst[32] = { 0 };
```

Comment:
This "32" is, I guess, the maximum size of the xattr. Maybe is good Idea to create a `#define [name] 32` to avoid errors, because this value is used below